lib/arqo/query.rb

```diff
@@ -12,12 +12,43 @@ def initialize(relation = derived_relation)
 
     private
 
-    def derived_relation_name
-      self.class.name.sub(/Query$/, '')
+    # Returns the model to which the query object is associated.
+    # By default it will infer the name from the query object's class
+    # assuming the name is the name of the model with "Query" as the suffix.
+    #
+    # As an example, UserQuery would be associated with the User model and
+    # Blog::PostQuery would be associated with the Blog::Post namespaced model.
+    #
+    # This method can be overridden to bypass this convention and associate
+    # the query object to another relation by returning it.
+    #
+    # ==== Example
+    #     class PostQuery < ARQO::Query
+    #       # ...
+    #
+    #       private
+    #
+    #       def associated_relation
+    #         Blog::Post
+    #       end
+    #     end
+    #
+    # @return [Class, ActiveRecord::Relation]
+    def associated_relation
+      class_name = self.class.name
+      derived_relation_name = class_name.sub(/Query$/, '')
+
+      unless Object.const_defined?(derived_relation_name)
+        raise NameError, "Could not find model #{derived_relation_name} associated " \
+          "to query #{class_name}.\n Make sure the name is correct or override " \
+          '#associated_relation to provide a custom model'
+      end
+
+      derived_relation_name.constantize
     end
 
     def derived_relation
-      derived_relation_name.constantize.all if Object.const_defined?(derived_relation_name)
+      associated_relation.all
```

Comment:
maybe you can use `associated_relation.all` directly in https://github.com/rootstrap/arqo/pull/11/files#diff-8951b5e34e4c0c507144be1892ee1384R9 and remove this method?